src/util/util.c

```diff
@@ -589,6 +589,14 @@ errno_t sss_filter_sanitize_ex(TALLOC_CTX *mem_ctx,
             output[j++] = '5';
             output[j++] = 'c';
             break;
+        case '\n':
+            output[j++] = '\\';
+            output[j++] = '0';
+            output[j++] = 'd';
+            output[j++] = '\\';
```

Comment:
Did you try without "bad" at the end?  The same as in ticket description.

sudo sss_cache -E; getent passwd $(echo -e "ad1\r")

If it works for you then we needn't include '\r'

BTW ATM I cannot reproduce in 1.15.1 but I assume it is some bug after cache_req refactoring